x-pack/plugins/enterprise_search/server/routes/enterprise_search/indices.ts

```diff
@@ -989,4 +994,165 @@ export function registerIndexRoutes({
       }
     })
   );
+
+  router.post(
+    {
+      path: '/internal/enterprise_search/ml/models/{modelName}',
+      validate: {
+        params: schema.object({
+          modelName: schema.string(),
+        }),
+      },
+    },
+    elasticsearchErrorHandler(log, async (context, request, response) => {
+      const modelName = decodeURIComponent(request.params.modelName);
+      const {
+        elasticsearch: { client },
+        savedObjects: { client: savedObjectsClient },
+      } = await context.core;
+      const trainedModelsProvider = ml
+        ? await ml.trainedModelsProvider(request, savedObjectsClient)
+        : undefined;
+
+      const savedObjectService = mlSavedObjectServiceFactory(
+        savedObjectsClient,
+        savedObjectsClient,
+        true,
+        undefined,
+        client,
+        () => Promise.resolve() // pretend isMlReady, to allow us to initialize the saved objects
```

Comment:
question: this feels a bit icky, what if ml isn't ready?

Reply:
This was actually lifted from some ML-side code itself... but, if we are able to call this function, the ML module _will_ be guaranteed to be ready, so I think it's OK to leave this here (although I'll remove the comment to lessen confusion ;) )